hybrid/Makefile.common

```diff
@@ -0,0 +1,50 @@
+# Set by platform-specific makefiles. For example: ~/cheri/output/sdk
+ifndef SDKBASE
+$(error SDKBASE is not set)
+endif
+
+# For example: morello-purecap
+ifndef PLATFORM
+$(error PLATFORM is not set)
+endif
+
+CC := $(SDKBASE)/bin/clang
+CFORMAT := $(SDKBASE)/bin/clang-format
+
+RUNUSER ?= root
+RUNHOST ?= localhost
+RUNDIR ?=
+CFLAGS ?=
+
+BINDIR := bin/$(PLATFORM)
+CFILES := $(wildcard *.c)
+HFILES := $(wildcard include/*.h)
+BINS := $(patsubst %.c,$(BINDIR)/%,$(CFILES))
+RUNTGTS := $(patsubst $(BINDIR)/%,run-%,$(BINS))
+
+.PHONY: all clang-format clean $(RUNTGTS)
+
+all: $(BINS)
+
+# Remove outputs, but only remove the top-level 'bin/' if it's empty. This
+# mitigates the risk of accidental damage, e.g. if run from a user's home when
+# they have a ~/bin.
+clean:
+	rm -rf $(BINDIR)
```

Comment:
This removes `BINDIR` so the comment above is redundant (also `rmdir --ignore-fail-on-non-empty` isn't portable, so we don't want to use that).

Reply:
This is from a past PR, copy/pasted and already approved as correct :)

Reply:
Alas, copy and paste does not prove anything is correct -- you'll need to prove to me that this is correct again (because I'm rather certain that the comment is completely incorrect, unless `rm -rf` magically doesn't delete a directory!).

Reply:
It was from my first makefiles PR (#36), but the latest one (#37) does a slightly better job.

Reply:
But that was not my PR, so I assumed was the preferred/approved way for a `Makefile.common`.

Reply:
@ltratt It's not perfect. It will, of course, delete your `./bin/morello-purecap` (or similar) directory, but that's rather less likely to be hit by accident than the `./bin` that we used to clean.

Reply:
Oh, hang on: `BINDIR` is a *subdir* of `bin/`. Hmm. That is a bit confusing! Perhaps it's not worth obsessing over, though. Let's not worry about this one anymore.